scripts/analysis.py

```diff
@@ -250,229 +116,198 @@ def get_sim_time_duration(cursor):
-
-    Returns
-    -------
-    list
-        isotope name list
-    list
-        isotope mass list
-    list
-        isotope transaction time list
-
-    """
-
-    temp_isotope = []
-    temp_mass = []
-    time_list = []
-    for res in resources:
-        res_qualid = res[2]
-        indices = [x for x, y in enumerate(compositions) if y[0] == res_qualid]
-        for index in indices:
-            nucid = compositions[index][1]
-            mass_frac = compositions[index][2]
-            mass_waste = res[0]
-            res_time = res[1]
-            temp_isotope.append(nucid)
-            temp_mass.append(mass_frac*mass_waste)
-            time_list.append(res_time)
-
-    return temp_isotope, temp_mass, time_list
-
-
-def plot_in_out_flux(cursor, facility, influx_bool, title, outputname):
-    """plots timeseries outflux from facility name in kg.
+def get_timeseries(in_list, duration, multiplyby, cumulative):
+    """ returns a timeseries list of a given data
 
     Parameters
     ----------
-    cursor: sqlite cursor
-        sqlite cursor
-    facility: str
-        facility name
-    influx_bool: bool
-        if true, calculates influx,
-        if false, calculates outflux
-    title: str
-        title of the multi line plot
-    outputname: str
-        filename of the multi line plot file
+    in_list: list
+        list of data to be created into timeseries
+        list[0] = time
+        list[1] = value, quantity
+    duration: int
+        duration of the simulation
+    multiplyby: int
+        integer to multiply the value in the list by for
+        unit conversion from kilograms
+    cumulative: boolean
+        determine whether returned timeseries should be
+        cumulative sum over time
 
     Returns
     -------
-
+    timeseries list of data
     """
-
-    cur = cursor
-    agent_ids = get_agent_ids(cur, facility)
-    if influx_bool is True:
-        resources = cur.execute(exec_string(agent_ids,
-                                            'transactions.receiverId',
-                                            'sum(quantity), time, qualid')
-                                + ' GROUP BY time, qualid').fetchall()
+    value = 0
+    value_timeseries = []
+    array = np.array(in_list)
+    if cumulative:
+        for i in range(0, duration):
+            if len(array) > 0:
+                value += sum(array[array[:, 0] == i][:, 1])
+            value_timeseries.append(value * multiplyby)
     else:
-        resources = cur.execute(exec_string(agent_ids,
-                                            'transactions.senderId',
-                                            'sum(quantity), time, qualid')
-                                + ' GROUP BY time, qualid').fetchall()
-
-    compositions = cur.execute('SELECT qualid, nucid, massfrac\
-                               FROM compositions').fetchall()
+        for i in range(0, duration):
+            value = sum(array[array[:, 0] == i][:, 1])
```

Comment:
cumsum() can sum along any axis. I strongly recommend numpy arrays rather than lists of lists.